### PR TITLE
Fix potential deadlock when loading feature flags

### LINKED ISSUE #38009: Initializing MB from a config file is broken
### Describe the bug

If you start up a new instance of Metabase with `MB_CONFIG_FILE_PATH` set, and `MB_PREMIUM_EMBEDDING_TOKEN` set to a token that provides the `config-text-file` feature, it will nevertheless error with:
```
Execution error (ExceptionInfo) at metabase-enterprise.advanced-config.file/initialize! (file.clj:261).
Metabase config files require a Premium token with the :config-text-file feature.
`````

### To Reproduce

1. Set `MB_CONFIG_FILE_PATH` as described [here](https://www.metabase.com/docs/latest/configuring-metabase/config-file)
2. Set `MB_PREMIUM_EMBEDDING_TOKEN` to a token with full features
3. Start up Metabase with a fresh app DB

### Expected behavior

Should not error

### Logs

_No response_

### Information about your Metabase installation

```JSON
Current master
```

### Severity

unsure, feature might be totally busted

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by crisptrutski**
Fixes https://github.com/metabase/metabase/issues/38009

### Description

This makes the feature flag loading code more robust by making the function re-entrant. To understand how this function could deadlock before, see the description in https://github.com/metabase/metabase/pull/38027.

It works by ensuring that the call to initialize the nonce comes from the same thread that initially locked on fetching the tokens.

There is a subtle behavior change here - was allow for a 10s timeout on each call individually now. Since it is unlikely that both endpoints will timeout consecutively, and the combined wait time is still fairly short, I think this is OK.

### How to verify

You can use the same steps as https://github.com/metabase/metabase/pull/38027, either of these PRs are sufficient to fix the bug.

### CODE REVIEW

File: src/metabase/public_settings/premium_features.clj [hunk 1]
```diff
@@ -117,32 +130,26 @@
...
-                                      :error-details (.getMessage e1)})))))))
-          result (deref fut fetch-token-status-timeout-ms ::timed-out)]
-      (if (= result ::timed-out)
-        (do
-          (future-cancel fut)
-          {:valid         false
-           :status        (tru "Unable to validate token")
-           :error-details (tru "Token validation timed out.")})
-        result))))
+    (let [site-uuid (setting/get :site-uuid-for-premium-features-token-checks)]
```
Comment (qnkhuat) [inline src/metabase/public_settings/premium_features.clj:133]: I'd add a comment here on why we want this here, you could link it to the other PR's description you have.

**Review by qnkhuat — APPROVED** [review]
The fix works and it makes sense to me, so I'm approving.

It'd be nice if you can find a way to write a test for this, it'd be hard I think so it's up to you.

**Comment by crisptrutski** [comment]
### Merge activity

* **Jan 23, 8:59 AM**: @crisptrutski started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/38029).

**Comment by noahmoss** [comment]
@crisptrutski Does this need to be backported?

**Comment by crisptrutski** [comment]
> @crisptrutski Does this need to be backported?

Strictly speaking we'll be OK with just https://github.com/metabase/metabase/pull/38027 being backported, but I'll sleep better if this is backported as well. Will try run the github action again.

**Pull request merged**